### DISCUSSION TITLE
Where to add plugin partials?

### DISCUSSION THREAD

**Discussion by tarunhere** [wintercms/winter#381] [Q&A] [answered]
I was referencing to Location plugin from October to make slightly modified version in wintercms. I have added the partials in `modulename/components/partials/state-district.htm` but when using it on frontend component of another plugin is throwing error of partial not found.

I checked the plugin page of WinterCMS and I don't see components folder and I tried with that change as well but still it shows partial not found error.

I need to use the dropdown of state and district in other forms but not able to make it work. I had added the behavior as well in location module. Any pointer will be highly appreciated. Thanks in advance.

**Comment by LukeTowers** [comment, marked as answer]
I'm not sure what you're asking, can you clarify what it is you're trying to do exactly? Perhaps https://wintercms.com/docs/cms/components#customizing-default-markup is what you're looking for?

**Reply by tarunhere** [reply]
Let me start from scratch.

I am making a plugin. I have a requirement to get state and district as a dropdown (just like country and state).
Now I have made a partial which has dropdown for state and district where district dependsOn state selection. I am trying to use this plugin's state-district.htm partial inside another plugin like this - `{% partial "state-district" %}`  and it shows me error - "The partial 'state-district' is not found."

Now my question is where shall we put our plugin's partials. I have tried keeping them in ``modulename/components/partials/state-district.htm`` and ``modulename/partials/state-district.htm`` but still it shows error. 
I think we don't need to register the partials inside Plugin.php.

Need your direction here. I hope to make sense now. Sorry for the earlier incomplete query.

**Reply by LukeTowers** [reply]
First problem, you probably shouldn't be trying to render another plugin's partials from within your own plugin. 

If this is for a custom form field (for the backend), it sounds like you might be better off having your first plugin [create](https://wintercms.com/docs/console/scaffolding#scaffold-create-formwidget) and [register](https://wintercms.com/docs/backend/widgets#form-widget-registration) a [custom FormWidget](https://wintercms.com/docs/backend/widgets#form-widgets) and then reference that FormWidget from within your second plugin.

Second, if you really must render an external (backend) partial from within your plugin, you should be able to reference it using [path symbols](https://wintercms.com/docs/services/helpers#path-symbols), so `<?= $this->makePartial('$/externalauthor/externalplugin/partials/example.htm'); ?>`

Finally, getting back to your original question, components within the same plugin can [share partials](https://wintercms.com/docs/plugin/components#component-partials), but I'm not sure if that would extend to partials from other plugins, I'm thinking probably not. Plugins can depend on other plugins for APIs, services, data, but I don't think it's a good idea for one plugin to have a dependency on another plugin's frontend view files to render said data; and is probably a symptom that you could be accomplishing your goals in a different, more simple and less fragile way.

**Reply by tarunhere** [reply]
Thanks for your reply. I really appreciate it. To clarify your doubts, I need to use these dropdowns (of state & district) on Front-end forms and not backend.
I am still confused as to how can I use this in frontend. One solution I found is to add the form functions (markup tags), I guess that will help me but how to add relationship of dependson for district dropdown. I am still searching... (can u guide me some doc here)

Basically I need to show this dropdowns just to fetch values in frontend form.

**Reply by tarunhere** [reply]
I think this image will give more idea. I want two dropdowns where value of second depends on first.
![address-state-district](https://user-images.githubusercontent.com/4122496/146669141-1cc13176-12d8-4248-8b54-7a7a845b9224.png)

**Reply by AIC-BV** [reply]
I think you want to use an Ajax event?
https://wintercms.com/docs/ajax/handlers

You will need to use 'onChange' instead of 'onSubmit'.
So when the value changes, you will trigger an ajax event returning the districts.
Then you can insert those districts with JS/Jquery or with twig by updating the partial (and placing the current districts in a partial: https://wintercms.com/docs/ajax/update-partials)

**Reply by tarunhere** [reply]
Thanks @AIC-BV for the pointers. I was not able to find a proper solution but this will definitely help.
I will follow your points and I am sure I will get desired output.
Take care
